tests/unit/s2n_server_new_session_ticket_test.c

```diff
@@ -24,85 +26,101 @@
 #define TICKET_AGE_ADD_MARKER sizeof(uint8_t)  + /* message id  */ \
                               SIZEOF_UINT24    + /* message len */ \
                               sizeof(uint32_t)   /* ticket lifetime */
+#define RECORD_LEN_MARKER     sizeof(uint8_t) +  /* message type */ \
+                              sizeof(uint16_t)   /* protocol version */
+
+static int s2n_setup_test_keys(struct s2n_connection *conn, struct s2n_config *config)
+{
+    notnull_check(conn);
+    notnull_check(config);
+
+    /**
+     *= https://tools.ietf.org/rfc/rfc5869#appendix-A.1
+     *# PRK  = 0x077709362c2e32df0ddc3f0dc47bba63
+     *#        90b6c73bb50f9c3122ec844ad7c2b3e5 (32 octets)
+     **/
+    S2N_BLOB_FROM_HEX(ticket_key,
+    "077709362c2e32df0ddc3f0dc47bba63"
+    "90b6c73bb50f9c3122ec844ad7c2b3e5");
+
+    /**
+     *= https://tools.ietf.org/rfc/rfc8448#section-3
+     *# secret (32 octets):  18 df 06 84 3d 13 a0 8b f2 a4 49 84 4c 5f 8a
+     *# 47 80 01 bc 4d 4c 62 79 84 d5 a4 1d a8 d0 40 29 19
+     **/
+    S2N_BLOB_FROM_HEX(test_master_secret,
+    "18df06843d13a08bf2a449844c5f8a"
+    "478001bc4d4c627984d5a41da8d0402919");
+
+    /* Set up encryption key */
+    uint64_t current_time;
+    uint8_t ticket_key_name[16] = "2016.07.26.15\0";
+    EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(config, 1));
+    EXPECT_SUCCESS(config->wall_clock(config->sys_clock_ctx, &current_time));
+    EXPECT_SUCCESS(s2n_config_add_ticket_crypto_key(config, ticket_key_name, strlen((char *)ticket_key_name),
+                    ticket_key.data, sizeof(ticket_key), current_time/ONE_SEC_IN_NANOS));
```

Comment:
I'm confused.

`sizeof(ticket_key)` is returning the size of an `s2n_blob` object, This is [defined here](https://github.com/aws/s2n-tls/blob/c72576f6fdc676d0b747ae9add033829c1d3e5c8/utils/s2n_blob.h#L23-L39) and looks completely unrelated to the size of `ticket_key.data`. Shouldn't you be passing in `ticket_key.size` rather than `sizeof(ticket_key)`?